frontend/src/components/shared/curies/CurieConceptSelector.jsx

```diff
@@ -0,0 +1,189 @@
+import React, { useRef, useEffect } from 'react';
+
+import ReactDOM from 'react-dom';
+
+import {
+  FormControl, Button, Badge, InputGroup,
+} from 'react-bootstrap';
+import shortid from 'shortid';
+
+import { AutoSizer, List } from 'react-virtualized';
+
+import entityNameDisplay from '@/utils/entityNameDisplay';
+import getNodeTypeColorMap from '@/utils/colorUtils';
+import curieUrls from '@/utils/curieUrls';
+
+export default function CurieConceptSelector({
+  concepts,
+  curies,
+  selection, handleSelect,
+  searchTerm, updateSearchTerm,
+  loading,
+  rightButtonFunction, rightButtonContents,
+  focus, clearFocus,
+}) {
+  // Reference to the input element used for setting focus
+  const inputRef = useRef(null);
+  useEffect(() => {
+    if (!focus) return;
+    // We have to use the deprecated find-dom-node method
+    // because react-bootstrap doesn't expose an alternative interface
+
+    // eslint-disable-next-line react/no-find-dom-node
+    const node = ReactDOM.findDOMNode(inputRef.current);
+    node.focus();
+  }, [focus]);
+
+  // After we have set focus, clear the variable so we can set it again
+  useEffect(() => { if (focus && clearFocus) clearFocus(); }, [focus]);
```

Comment:
This logic can go in the above useEffect. Also, I don't think we should need this logic. We should be able to simply call focus() on an input and forget about it, not go and reset some property.

Reply:
This logic is necessary if we want to be able to set focus again after setting it once. If we set focus to true again, it won't call the useEffect without this logic.

Reply:
I think what I'm getting at is that now we aren't passing refs, which is good, but now we have this other random property that we have to pass around. The reason I initially decided to duplicate the curie container logic was for this reason, that all the logic would be contained in one component and simpler to follow.

Reply:
I think that the decision to duplicate the curie container logic made sense when the curie container was simple and only pulled the curies from a JSON file. However, the number of lines (and complexity) in the NodePanel and useNode files has nearly doubled since then, so I think it now makes sense to try to re-use that logic as best we can.